src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkCommand.java

```diff
@@ -0,0 +1,79 @@
+package co.uk.cloudam.streaming.template.discord.link;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+import co.uk.cloudam.streaming.template.discord.SlashCommand;
+import discord4j.core.event.domain.interaction.ChatInputInteractionEvent;
+import lombok.extern.apachecommons.CommonsLog;
+import reactor.core.publisher.Mono;
+
+@Component
+@CommonsLog
+public class LinkCommand implements SlashCommand {
+
+    private final LinkCodeRepo linkCodeRepo;
+
+    @Autowired
+    public LinkCommand(LinkCodeRepo linkCodeRepo) {
+        this.linkCodeRepo = linkCodeRepo;
+    }
+
+    @Override
+    public String getName() {
+        return "link";
+    }
+
+    @Override
+    public Mono<Void> handle(ChatInputInteractionEvent event) {
+        var newToken = UUID.randomUUID();
+        var discordName = event.getInteraction().getUser().getUsername();
+        if (doNew(discordName)) {
+            addToDatabase(newToken, discordName);
+        }
+        Optional<LinkCodeEntity> searchByUser = linkCodeRepo.findById(discordName);
+        return event.reply()
+            .withEphemeral(true)
+            .withContent(searchByUser.isEmpty() ? "Sorry " + discordName + ", We couldn't get your token. " : "Hello " + searchByUser.get().getUserName() + ", " + searchByUser.get().getToken() + " .. -Insert nifty bow, here-");
+
+    }
+
+    private boolean doNew(String discordName) {
+        Optional<LinkCodeEntity> dataBase = linkCodeRepo.findById(discordName);
+        return dataBase.isEmpty() || isExpired(dataBase.get().getExpireTime());
+    }
+
+    private boolean isExpired(LocalDateTime expireTime) {
+        return expireTime == null || LocalDateTime.now().isAfter(expireTime);
```

Comment:
as we control all the data you can assume expreTime is never null (you may have to wipe you dev DB though)

Reply:
It was just a handler for those that existed in the database first. before the isExpired Script. Granted there isn't any currently. Fixed.